ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+TabWebDelegate.swift

```diff
@@ -0,0 +1,577 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveShared
+import Foundation
+import Preferences
+import Shared
+import Strings
+import UIKit
+
+/// A protocol that tells an object about web UI related events happening
+///
+/// `WKWebView` specific things should not be accessed from these methods, if you need to access
+/// the underlying web view, you should only access it via `Tab`
+protocol TabWebDelegate: AnyObject {
+  func tabWebViewDidClose(_ tab: Tab)
+  func tab(
+    _ tab: Tab,
+    contextMenuConfigurationForLinkURL linkURL: URL?
+  ) async -> UIContextMenuConfiguration?
+  func tab(
+    _ tab: Tab,
+    requestMediaCapturePermissionsFor type: WebMediaCaptureType
+  ) async -> WebPermissionDecision
+  func tab(_ tab: Tab, runJavaScriptAlertPanelWithMessage message: String, pageURL: URL) async
+  func tab(
+    _ tab: Tab,
+    runJavaScriptConfirmPanelWithMessage message: String,
+    pageURL: URL
+  ) async -> Bool
+  func tab(
+    _ tab: Tab,
+    runJavaScriptConfirmPanelWithPrompt prompt: String,
+    defaultText: String?,
+    pageURL: URL
+  ) async -> String?
+
+  func tab(
+    _ tab: Tab,
+    didRequestHTTPAuthFor protectionSpace: URLProtectionSpace,
+    proposedCredential credential: URLCredential?,
+    previousFailureCount: Int
+  ) async -> URLCredential?
+}
+
+/// Media device capture types that a web page may request
+enum WebMediaCaptureType {
+  case camera
+  case microphone
+  case cameraAndMicrophone
+}
+
+/// Permission decisions for responding to various permission prompts
+enum WebPermissionDecision {
+  case prompt
+  case grant
+  case deny
+}
+
+extension TabWebDelegate {
+  func tabWebViewDidClose(_ tab: Tab) {}
+
+  func tab(
+    _ tab: Tab,
+    contextMenuConfigurationForLinkURL linkURL: URL?
+  ) async -> UIContextMenuConfiguration? {
+    return nil
+  }
+
+  func tab(
+    _ tab: Tab,
+    requestMediaCapturePermissionsFor type: WebMediaCaptureType
+  ) async -> WebPermissionDecision {
+    return .prompt
+  }
+
+  func tab(_ tab: Tab, runJavaScriptAlertPanelWithMessage message: String, pageURL: URL) async {}
+
+  func tab(
+    _ tab: Tab,
+    runJavaScriptConfirmPanelWithMessage message: String,
+    pageURL: URL
+  ) async -> Bool {
+    return false
+  }
+
+  func tab(
+    _ tab: Tab,
+    runJavaScriptConfirmPanelWithPrompt prompt: String,
+    defaultText: String?,
+    pageURL: URL
+  ) async -> String? {
+    return
+      nil
```

Comment:
Formatter being weird? `nil` might not need to be on a new line.

Reply:
Nah I must have accidentally hit return and not seen, this and other formatting related comments fixed in https://github.com/brave/brave-core/pull/27746/commits/536eb4b1dd31309d2f9816a4cb35a68476f8ed55